pyfr/writers/vtk.py

```diff
@@ -9,18 +9,60 @@
 from pyfr.shapes import BaseShape
 from pyfr.util import memoize, subclass_where
 from pyfr.writers import BaseWriter
+from pyfr.writers.nodemaps import VTKHONodeMaps
 
 
 class VTKWriter(BaseWriter):
     # Supported file types and extensions
     name = 'vtk'
     extn = ['.vtu', '.pvtu']
 
+    vtk_types_ho = dict(tri=69, quad=70, tet=71, pri=73, hex=72)
+
     def __init__(self, args):
         super().__init__(args)
 
         self.dtype = np.dtype(args.precision).type
-        self.divisor = args.divisor or self.cfg.getint('solver', 'order')
+
+        # Divisor for each type element
+        self.etypes_div = defaultdict(lambda: self.divisor)
+
+        # Choose whether to output subdivided cells or high order VTK cells
+        # If -k is given by the user then use high-order VTK cells as output
+        # with order equal to the solution order or to the one provided by
+        # the user
+        # Else if neither -o nor -d are found in the input then use high-order
+        # VTK cells with order equal to the simulation order
+        # Else use cell subdivision
+        if args.order:
```

Comment:
args.order or args.divisor is None

Reply:
I see now, yes it's true that it can be better organized. Let me know if the new commit works better for you.